src/twisted/conch/test/test_transport.py

```diff
@@ -515,22 +515,60 @@ def test_sendVersion(self):
         )
         self.assertRegex(softwareVersion, softwareVersionRegex)
 
+    def test_dataReceiveVersionNullCharacter(self):
+        """
+        It will reject the connection when null characters are found in the
+        peer's SSH version string.
+        """
+        sut = MockTransportBase()
+        sut.makeConnection(self.transport)
+
+        # White the new line is not received, all looks fine so far.
+        sut.dataReceived(b"SSH-2-to-be-continued-")
+        self.assertFalse(self.transport.disconnecting)
+
+        # Once the new line is received the version string is validated.
+        sut.dataReceived(b"with-\x00-character\r\nmore-data-ignored")
+
+        self.assertTrue(self.transport.disconnecting)
+        self.assertIn(
+            b"Invalid peer version. Null character found.", self.transport.value()
+        )
+
+    def test_dataReceiveVersionInvalidVersionFormat(self):
+        """
+        It will reject the connection when remote peer SSH format doesn't start
+        with `SSH-`.
+        """
```

Comment:
Can you make this test server-specific, and add a client-specific test that ensures that lines before the identification string are allowed?

Reply:
Yes. Will do.